### PR TITLE
chore(ddtags): capture key before first colon

### PR DESCRIPTION
**Pull request by mabdinur**
### Description

Addresses edge cases missing from: https://github.com/DataDog/dd-trace-php/pull/3162. 
Failing tests: https://github.com/DataDog/system-tests/actions/runs/14201484499/job/39789366220?pr=4439 

## Changes 

- The new parser is currently not used. We need to load `DD_TAGS` using `CUSTOM(MAP)`
- Ensure the first `:` colon in a tag is used to split key-value pairs. 
- If there are no colons in tag then the full tag becomes the key. 
- If there are more than one colons in a tag then the remaining colons must be included in the value.

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -4063,7 +4063,7 @@ jobs:
             docker-compose --version
       - run:
           name: Clone System Tests repo
-          command: git clone https://github.com/DataDog/system-tests.git
+          command: git clone https://github.com/DataDog/system-tests.git && git checkout munir/enable-php-ddtags-tests
```
Comment (mabdinur) [inline .circleci/continue_config.yml:4066]:
```suggestion
          command: git clone https://github.com/DataDog/system-tests.git
```

**Comment by mabdinur** [comment]
This implementation fails the following the following test cases:

```
    "key1:value1 key2:value2": [("key1", "value1"), ("key2", "value2")], 
    "env:test aKey:aVal bKey:bVal cKey:": [("env", "test"), ("aKey", "aVal"), ("bKey", "bVal"), ("cKey", "")],
    "env:test     bKey :bVal dKey: dVal cKey:": [
        ("env", "test"),
        ("bKey", ""),
        ("dKey", ""),
        ("dVal", ""),
        ("cKey", ""),
    ],
    "env:keyWithA:Semicolon bKey:bVal cKey": [("env", "keyWithA:Semicolon"), ("bKey", "bVal"), ("cKey", "")],
 ```

In the cases above `' ' `is never used  as the delimter.

**Review by bwoebi — APPROVED** [review]
